Automattic/sensei
# Adds a notice when adding the first Sensei block or removing the last one from a Course or Lesson.

Fixes #4242

### Changes proposed in this Pull Request

* Adds a notice when adding the first Sensei block or removing the last one from a Course or Lesson.
* Refactors the toggle legacy metaboxes, joining the course and lesson implementations.

### Testing instructions


* Make sure when removing/adding blocks to a course or lesson, the [respective metaboxes](https://github.com/Automattic/sensei/pull/4246/files#diff-83b8ab150f30205bd02efb5acb9d6636b5b729aeb4f82d9bc999606a7ee3c1cdR23-R32) are toggled (same behavior as previously).
* Make sure when you remove all the blocks, you see a warning that the page will render the template.
* Make sure when you add at least one block (after a state with no course/lesson blocks), you see a warning that the template will not be used anymore.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/125347952-fb108a80-e331-11eb-998c-7c0f20ac6196.mov
